pkg/apis/sources/config/pingconfig/config/ping_template_types.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// This is a verbatim copy from messaging/v1beta1/ping_template_types.go
+// but we can not import v1beta1 since we use this there as well, so
+// replicate it here.
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PingTemplateSpec struct {
```

Comment:
why is this needed?

Reply:
It generates the deepcopy for store.  Model pattern  as channel store

Reply:
This is quite confusing. For channel it's different because the config and the spec are the same. That's not the case for PingSource. I would remove this and just use `PingDefaults`. WDYT?

Reply:
I'll investigate that. Currently, preoccupied with something else will get back ASAP. Thanks